web/client/components/data/featuregrid/editors/DropDownEditor.jsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+*/
+const React = require('react');
+const PropTypes = require('prop-types');
+const AttributeEditor = require('./AttributeEditor');
+const {ComboboxWithForceSelection} = require('../../../misc/ComboboxWithEnhancer');
+const {head} = require('lodash');
+const assign = require('object-assign');
+const {forceSelection} = require('../../../misc/enhancers/basic');
+
+class DropDownEditor extends AttributeEditor{
+    static propTypes = {
+        column: PropTypes.object,
+        dataType: PropTypes.string,
+        defaultOption: PropTypes.string,
+        forceSelection: PropTypes.bool,
+        allowEmpty: PropTypes.bool,
+        inputProps: PropTypes.object,
+        isValid: PropTypes.func,
+        onBlur: PropTypes.func,
+        typeName: PropTypes.string,
+        url: PropTypes.string,
+        value: PropTypes.string,
+        values: PropTypes.array
+    };
+    static defaultProps = {
+        isValid: () => true,
+        dataType: "string",
+        values: [],
+        forceSelection: true,
+        allowEmpty: true
+    };
+    constructor(props) {
+        super(props);
+        this.validate = (value) => {
+            try {
+                return this.props.isValid(value[this.props.column && this.props.column.key]);
+            } catch (e) {
+                return false;
+            }
+        };
+        this.getValue = () => {
+            const updated = super.getValue();
+
+            if (this.props.forceSelection) {
+                return {[this.props.column.key]: forceSelection({
+                    oldValue: this.props.defaultOption,
+                    changedValue: updated[this.props.column && this.props.column.key],
+                    data: this.props.values,
+                    allowEmpty: this.props.allowEmpty})};
+            }
+            if (this.props.allowEmpty) {
+                return updated;
+            }
+            // this case is only when forceSelection and allowEmpty are falsy, but this is contractidtory!! so the default option is used
+            return {[this.props.column.key]: forceSelection({
```

Comment:
Do you agree with this implementation?
in my opinion forceSelection = false && allowEmpty = false does not make sense. 
in that case i'll do the following: 
forceselection becomes true, allow empty stay false.

opinions?